### DISCUSSION TITLE
How to use workflow files only from a specific branch only?

### DISCUSSION THREAD

**Discussion by aahnik** [community/community#25465] [Actions]
<p>I have a GitHub action that does something whenever a new issue is created.</p>
<p>Now the workflow yaml file is in dev branch.</p>
<p>And after pushing the yaml, file the action started.</p>
<p>I want actions to run from workflow files only in main branch.</p>
<p>Is there any settings option, to disable workfile file run from a specific branch?</p>
<p>Basically what I want is, workflow files which are in allowed branches, should run.</p>
<p>I am not talking about on : triggers.</p>
<p>I am talking about a general repo setting, that would enforce github to ignore any workflow files in branches other than allowed branches</p>

**Comment by Simran-B** [comment, marked as answer]
<p>You can restrict what actions are allowed in workflows, but there is no repository setting to restrict on what branches workflows are allowed to run:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository" target="_blank" rel="noopener nofollow ugc">Disabling or limiting GitHub Actions for a repository - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You could enable <em>Require status checks to pass before merging</em> and only let actions succeed if the branch is on an allow list, but that will have to be coded into the workflow itself.</p>
